### PR TITLE
SDK-2075 SDK-2085 Add subject and identity profile requirements

### PR DESCRIPTION
**Pull request by LuborRod**

### CODE REVIEW

File: tests/ShareUrl/DynamicScenarioBuilderTest.php
Comment (echarrod) [inline tests/ShareUrl/DynamicScenarioBuilderTest.php:77]: I think it's best to be explicit here, and include this null value (like you've done), rather than omitting it. @BenSmithYoti I think this matches the .NET implementation, right?

**Review by echarrod — APPROVED** [review]
All good from my side, would be good to get @BenSmithYoti's approval before merging :+1:

**Review by BenSmithYoti — APPROVED** [review]
Responded to Ed's comment. All looks good to me. Thanks Rodion. Perfect!

File: tests/ShareUrl/DynamicScenarioBuilderTest.php
Comment (BenSmithYoti) [reply tests/ShareUrl/DynamicScenarioBuilderTest.php:77]: Another e.g. is `issuing_authority_sub_check ` on `RequestedDocumentAuthenticityConfig`  

In short I believe the implementation in this PR will send up the following JSON: `"identity_profile_requirements" : null,` and I believe this will be handled fine by the API (same behaviour as if identity_profile_requirements was excluded from the JSON).

**Pull request merged**
